src/Microsoft.OData.Client/Materialization/ODataMaterializerContext.cs

```diff
@@ -15,15 +15,27 @@ namespace Microsoft.OData.Client.Materialization
     /// </summary>
     internal class ODataMaterializerContext : IODataMaterializerContext
     {
+        private bool includeLinks;
+
         /// <summary>
         /// Initializes a materializer context
         /// </summary>
         /// <param name="responseInfo">Response information used to initialize with the materializer</param>
         /// <param name="materializerCache">The materializer cache.</param>
-        internal ODataMaterializerContext(ResponseInfo responseInfo, MaterializerCache materializerCache)
+        /// <param name="includeLinks">Whether to include navigation properties when materializing an entry.</param>
+        internal ODataMaterializerContext(ResponseInfo responseInfo, MaterializerCache materializerCache, bool includeLinks = true)

```

Comment:
should the default `includeLinks` value here be true or false?

Reply:
The default behaviour is `true`.
For deep insert, we want the behaviour to be `false` since for each level, we don't need to include links since they will be processed in the next level.

Reply:
I think this should be added for deep update also